### PR TITLE
Updates NFT tutorial to use latest best practices

### PR DESCRIPTION
**Pull request by joshuahannan**
Closes https://github.com/onflow/flow/issues/627

## Description

Updates the NFT tutorial to use the latest terminology and best practices in Cadence
Updates the names of some contracts, transactions, and scripts
Directs users to useful production examples and metadata standard

______

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: docs/tutorial/04-non-fungible-tokens.mdx [hunk 1]
```diff
@@ -321,33 +347,60 @@ pub contract ExampleNFT {
...
 
 	init() {
+        // use unique path names
+        self.CollectionStoragePath = /storage/nftTutorialCollection
+        self.CollectionPublicPath = /public/nftTutorialCollection
+        self.MinterStoragePath = /storage/nftTutorialMinter
+
 		// store an empty NFT Collection in account storage
-        self.account.save(<-self.createEmptyCollection(), to: /storage/NFTCollection)
+        self.account.save(<-self.createEmptyCollection(), to: self.CollectionStoragePath)
```
Comment (bjartek) [inline docs/tutorial/04-non-fungible-tokens.mdx:356]: I think it is confusing that this is in the contract code. For all users except the user that deploys the contract this has to be done in a transaction. So why not learn people that in this tutorial?

File: docs/design-patterns.mdx [hunk 2]
```diff
@@ -2,9 +2,13 @@
...
 
-This is a selection of software design patterns developed by core Flow developers while writing Cadence code for deployment to Flow Mainnet.
+This is a selection of software design patterns developed by core Flow developers
+while writing Cadence code for deployment to Flow Mainnet.
 
-[Design patterns](https://en.wikipedia.org/wiki/Software_design_pattern) are building blocks for software development. They may provide a solution to a problem that you encounter when writing smart contracts in Cadence. Remember that if they do fit they may not be the right solution for a given situation, and they are not meant to be a set of prescriptions to be followed exclusively even where a better solution represents itself.
+[Design patterns](https://en.wikipedia.org/wiki/Software_design_pattern) are building blocks for software development.
+They may provide a solution to a problem that you encounter when writing smart contracts in Cadence.
+Remember that if they do fit they may not be the right solution for a given situation,
+and they are not meant to be a set of prescriptions to be followed exclusively even where a better solution represents itself.
```
Comment (gregsantos) [inline docs/design-patterns.mdx:11]: copy edit as this feels clunky to me. How about:

_If they do not clearly fit, these patterns may not be the right solution for a given situation or problem.
They are not meant to be rules to be followed strictly, especially where a better solution presents itself._

File: docs/tutorial/03-fungible-tokens.mdx [hunk 3]
```diff
@@ -512,7 +512,7 @@ Capabilities allow us to accomplish this safely.
 
 ---
 
-Another important feature in Cadence is its utilization of **Capability Security.**
+Another important feature in Cadence is its utilization of [**Capability Security.**](/cadence/language/capability-based-access-control)
 This feature ensures that, while the withdraw function is declared public on the resource,
 no one except the intended user and those they approve of can withdraw tokens from their vault.
```
Comment (gregsantos) [inline docs/tutorial/03-fungible-tokens.mdx:517]: Removes unneeded comma

_This feature ensures that while the withdraw function is declared public on the resource,
no one except the intended user and those they approve of can withdraw tokens from their vault._

File: docs/tutorial/04-non-fungible-tokens.mdx (hunk 1 above)
Comment (joshuahannan) [reply docs/tutorial/04-non-fungible-tokens.mdx:356]: We do teach that in this tutorial in a transaction that we use later for account 0x01. It is just convenient to have it here also

**Review by turbolent — APPROVED** [review]
Looks good!

**Pull request merged**